lsif-semanticdb/src/main/java/com/sourcegraph/lsif_semanticdb/SignatureFormatter.java

```diff
@@ -0,0 +1,309 @@
+    if (signature.hasClassSignature()) {
+      formatClassSignature(signature.getClassSignature());
+    } else if (signature.hasMethodSignature()) {
+      formatMethodSignature(signature.getMethodSignature());
+    } else if (signature.hasValueSignature()) {
+      formatValueSignature(signature.getValueSignature());
+    } else if (signature.hasTypeSignature()) {
+      formatTypeParameterSignature(signature.getTypeSignature());
+    }
+
+    return s.toString();
+  }
+
+  private void formatClassSignature(ClassSignature classSignature) {
+    boolean isAnnotation =
+        classSignature.getParentsList().stream()
+            .anyMatch(t -> t.getTypeRef().getSymbol().equals(ANNOTATION_SYMBOL));
+
+    printKeyword(formatAccess());
+    if (!has(Property.ENUM) && !isAnnotation) printKeyword(formatModifiers());
+
+    switch (symbolInformation.getKind()) {
+      case CLASS:
+        if (has(Property.ENUM)) {
+          printKeyword("enum");
+        } else {
+          printKeyword("class");
+        }
+        break;
+      case INTERFACE:
+        if (isAnnotation) {
+          printKeyword("@interface");
+          break;
+        }
+        printKeyword("interface");
+        break;
+    }
+    s.append(symbolInformation.getDisplayName());
+
+    List<SymbolInformation> typeParameters = getSymlinks(classSignature.getTypeParameters());
+    if (!typeParameters.isEmpty()) {
+      s.append(
+          typeParameters.stream()
+              .map(this::formatTypeParameter)
+              .collect(Collectors.joining(", ", "<", ">")));
+    }
+
+    List<Type> parentsList =
+        classSignature.getParentsList().stream()
+            .filter(parent -> !parent.equals(OBJECT_TYPE_REF))
+            .filter(parent -> !parent.getTypeRef().getSymbol().equals(ENUM_SYMBOL))
+            .filter(parent -> !parent.getTypeRef().getSymbol().equals(ANNOTATION_SYMBOL))
+            .collect(Collectors.toList());
+    // TODO: extends vs implements
+    if (!parentsList.isEmpty()) {
+      printKeyword(" extends");
+
+      String parents = parentsList.stream().map(this::formatType).collect(Collectors.joining(", "));
+      s.append(parents);
+    }
+  }
+
+  private void formatMethodSignature(MethodSignature methodSignature) {
+    printKeyword(formatAccess());
+    printKeyword(formatModifiers());
+
+    List<SymbolInformation> typeParameters = getSymlinks(methodSignature.getTypeParameters());
+    if (!typeParameters.isEmpty()) {
+      printKeyword(
+          typeParameters.stream()
+              .map(this::formatTypeParameter)
+              .collect(Collectors.joining(", ", "<", ">")));
+    }
+
+    printKeyword(formatType(methodSignature.getReturnType()));
+    s.append(symbolInformation.getDisplayName());
+
+    s.append(
+        methodSignature.getParameterListsList().stream()
+            .flatMap((params) -> getSymlinks(params).stream())
+            .map(
+                (symInfo) ->
```

Comment:
FYI, I don't think you need to wrap the parameter in parentheses
```suggestion
                symInfo ->
```